### DISCUSSION TITLE
Manually creating a compressed-payload JWE?

### DISCUSSION THREAD

**Discussion by jmandel** [panva/jose#666] [Q&A] [answered]
I understand that the jose library no longer supports automatically compressing payloads when a `"zip": "DEF"` header is set.

But currently the library also prevents me from manually deflating my payload and setting `"zip": "DEF"`  to indicate the deflated status for recipients. Is this intentional?

Would it be feasible to allow some flag or note to the "encrypt" API just saying "yes, I as the caller am going to handle deflation; no need to do anything but please don't blow up" ?

**Comment by panva** [comment, marked as answer]
You can use [v4.x](https://github.com/panva/jose#supported-versions) for at least the foreseeable future if you need to create compressed payloads.

I don't plan another major at least until `--experimental-require-module` is on by default in two LTS release lines and will continue supporting v4.x with security fixes should they be needed until then.

**Reply by jmandel** [reply]
Thanks for the response + workaround! Is it worth including this "ignore zip claim" flag as a requested feature for a subsequent release when 4.x is deprecated?
